x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -541,8 +544,27 @@ protected void doNextSearch(SearchRequest request, ActionListener<SearchResponse
 
         @Override
         protected void doNextBulk(BulkRequest request, ActionListener<BulkResponse> nextPhase) {
-            ClientHelper.executeWithHeadersAsync(transformConfig.getHeaders(), ClientHelper.DATA_FRAME_ORIGIN, client, BulkAction.INSTANCE,
-                    request, nextPhase);
+            ClientHelper.executeWithHeadersAsync(transformConfig.getHeaders(),
+                ClientHelper.DATA_FRAME_ORIGIN,
+                client,
+                BulkAction.INSTANCE,
+                request,
+                ActionListener.wrap(bulkResponse -> {
+                    if (bulkResponse.hasFailures() && auditBulkFailures.get()) {
+                        int failureCount = 0;
+                        for(BulkItemResponse item : bulkResponse.getItems()) {
+                            if (item.isFailed()) {
+                                failureCount++;
+                            }
+                        }
+                        auditor.warning(transformId,
+                            "Experienced at least [" +
+                                failureCount +
+                                "] bulk index failures. See the logs of the node running the transform for details.");
```

Comment:
It might be nice to also include the result of `bulkResponse.buildFailureMessage()` here as it might be enough to avoid having to look in logs. We can still prompt them to look in logs for details.

Reply:
@dimitris-athanasiou I initially thought of that too, but doing some experiments, if many of the bulk messages failed, the output could be huge. I can add it here for sure, I just worry about readability (I suppose its no less readable than the logs...)

Reply:
Not sure what the UI does currently, but it feels like it's a matter of the UI trimming the message and having a `show more` option if the user wants to expand and read the whole message. But many times the failure could be some silly misconfiguration for which the error message will be clear.